packages/preact/src/index.ts

```diff
@@ -351,6 +352,12 @@ export function useSignalEffect(cb: () => void | (() => void)) {
 	}, []);
 }
 
+export function useWatcher<T>(value: T) {
+	const watcher = useSignal(value);
+	watcher.value = value;
+	return watcher as ReadonlySignal<T>;
+}
```

Comment:
With this implementation, I sometimes get the following warning:

> react-dom.development.js:86 Warning: Cannot update a component (`Foo`) while rendering a different component (`Bar`). To locate the bad setState() call inside `Bar`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render

I changed it to this and the warning went away:
```ts
export default function useWatcher<T>(value: T): ReadonlySignal<T> {
  const watcher = useMemo<Signal<T>>(() => {
    return signal(value)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  useEffect(() => {
    watcher.value = value
  }, [value, watcher])
}
```

It's absolutely possible that the fault is entirely within my application and the change above papers over a real problem. It's also possible that setting `watcher.value = value` outside of a React hook is causing problems with React. I haven't narrowed that down yet.

Reply:
I think we should ignore it, its not something bad

Reply:
I know how to fix it: wrap component in batch, so it will not try to rerender instantly. What do you think? @andrewiggins